### PR TITLE
Fixes detection of OpenSSL's OCSP APIs

### PR DESCRIPTION
**Pull request by randall**
and cleans up logging around OCSP

### CODE REVIEW

**Comment by randall** [comment]
This should go into 8.0.0 as well

File: iocore/net/P_OCSPStapling.h [hunk 1]
```diff
@@ -21,13 +21,14 @@
...
 #pragma once
 
+#include "ts/ink_config.h"
+
+#if TS_USE_TLS_OCSP
 #include <openssl/ssl.h>
+#include <openssl/ocsp.h>
 
-#ifdef OCSP_sendreq_new
 #define HAVE_OPENSSL_OCSP_STAPLING 1
```
Comment (maskit) [inline iocore/net/P_OCSPStapling.h]: Should we keep HAVE_OPENSSL_OCSP_STAPLING? I think we can replace all HAVE_OPENSSL_OCSP_STAPLINGs with TS_USE_TLS_OCSP.

**Comment by SolidWallOfCode** [comment]
Looks good, but @maskit had a good comment.

File: iocore/net/P_OCSPStapling.h (hunk 1 above)
Comment (randall) [reply iocore/net/P_OCSPStapling.h]: Good idea! Updated patch.

**Comment by maskit** [comment]
[approve ci autest]

**Pull request merged**